.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,639 @@
+        required: false
+        type: string
+        default: ${{ vars.S3_BUCKET || 'resin-production-img-cloudformation' }}
+      aws-subnet:
+        description: AWS subnet for AMI deploys
+        required: false
+        type: string
+        default: ${{ vars.AWS_SUBNET || 'subnet-02d18a08ea4058574' }}
+      aws-security-group:
+        description: AWS security group for AMI deploys
+        required: false
+        type: string
+        default: ${{ vars.AWS_SECURITY_GROUP || 'sg-057937f4d89d9d51c' }}
+      sign-api-url:
+        description: URL for signing secure-boot images
+        required: false
+        type: string
+        default: ${{ vars.SIGN_API_URL || 'https://sign.balena-cloud.com' }}
+      yocto-cache-host:
+        description: Yocto NFS sstate cache host
+        required: false
+        type: string
+        default: ${{ vars.YOCTO_CACHE_HOST || 'nfs.product-os.io' }}
+      deploy-s3:
+        description: Whether to deploy images to S3
+        required: false
+        type: boolean
+        default: true
+      deploy-hostapp:
+        description: Whether to deploy a hostApp container image to a balena environment
+        required: false
+        type: boolean
+        default: true
+      deploy-ami:
+        description: Whether to deploy an AMI to AWS
+        required: false
+        type: boolean
+        default: false
+      sign-image:
+        description: Whether to sign image for secure boot
+        required: false
+        type: boolean
+        default: false
+      os-dev:
+        description: Enable OS development features
+        required: false
+        type: boolean
+        default: false
+      deploy-esr:
+        description: Enable to deploy ESR
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  build:
+    name: Build
+    runs-on: ${{ fromJSON(inputs.runs-on) }}
+    environment: ${{ inputs.github-environment }}
+
+    env:
+      MACHINE: "${{ inputs.machine }}"
+      automation_dir: "${{ github.workspace }}/balena-yocto-scripts/automation"
+      BALENARC_BALENA_URL: "${{ inputs.balena-environment }}"
+      WORKSPACE: ${{ github.workspace }}
+      VERBOSE: verbose
+      YOCTO_CACHE_DIR: /var/lib/yocto
+      BARYS_ARGUMENTS_VAR: ""
+
+    outputs:
+      hostapp_path: ${{ steps.build.outputs.hostapp_path }}
+      os_version: ${{ steps.build-lib.outputs.os_version }}
+      device_slug: ${{ steps.build-lib.outputs.device_slug }}
+      deploy_artifact: ${{ steps.build-lib.outputs.deploy_artifact }}
+      is_private: ${{ steps.build-lib.outputs.is_private }}
+      dt_arch: ${{ steps.build-lib.outputs.dt_arch }}
+      meta_balena_version: ${{ steps.build-lib.outputs.meta_balena_version }}
+      yocto_scripts_ref: ${{ steps.build-lib.outputs.yocto_scripts_ref }}
+
+    defaults:
+      run:
+        working-directory: .
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+
+    steps:
+      # https://github.com/actions/checkout
```

Comment:
We can probably avoid this clone if the build script lives with the device repository.

Reply:
All workflows need to perform a clone at the start, and since this is a shared workflow it will inherit the source repo and ref from the calling repository and not the inputs.